server/netty-server/zio/src/main/scala/sttp/tapir/server/netty/internal/NettyZioRequestBody.scala

```diff
@@ -44,17 +44,17 @@ private[netty] class NettyZioRequestBody[Env](createFile: ServerRequest => RIO[E
     }
   }
 
-  override def toStream(serverRequest: ServerRequest): streams.BinaryStream = {
-
-    serverRequest.underlying
+  override def toStream(serverRequest: ServerRequest, maxBytes: Option[Long]): streams.BinaryStream = {
+    val stream = serverRequest.underlying
       .asInstanceOf[StreamedHttpRequest]
       .toZIOStream()
       .flatMap(httpContent => ZStream.fromChunk(Chunk.fromByteBuffer(httpContent.content.nioBuffer())))
+    maxBytes.map(ZioStreams.limitBytes(stream, _)).getOrElse(stream)
   }
 
   private def nettyRequestBytes(serverRequest: ServerRequest): RIO[Env, Array[Byte]] = serverRequest.underlying match {
     case req: FullHttpRequest   => ZIO.succeed(ByteBufUtil.getBytes(req.content()))
-    case _: StreamedHttpRequest => toStream(serverRequest).run(ZSink.collectAll[Byte]).map(_.toArray)
+    case _: StreamedHttpRequest => toStream(serverRequest, maxBytes = None).run(ZSink.collectAll[Byte]).map(_.toArray) // TODO
```

Comment:
that's for the next PRs?

Reply:
Yes, coming soon :) https://github.com/softwaremill/tapir/pull/3337